src/linear_algebra/matrix/determinant.lean

```diff
@@ -260,6 +260,12 @@ lemma ring_hom.map_det {M : matrix n n R} {f : R →+* S} :
   f M.det = matrix.det (f.map_matrix M) :=
 by simp [matrix.det_apply', f.map_sum, f.map_prod]
 
+/-- Given a ring homorphism `f` from `S` to `R`, the induced ring homomorphism `f.map_matrix` from
+matrices over `S` to matrices over `R` preserves the property of having determinant 1. -/
+lemma ring_hom.map_det_one {M : matrix n n R} (f : R →+* S) (h : det M = 1) :
+  (f.map_matrix M).det = 1 :=
+by rw ← matrix.ring_hom.map_det; simp [h]
```

Comment:
This seems like a bit of an odd lemma to me. If it were to exist, the name would be something like `ring_hom.det_map_matrix_eq_one_of_det_eq_one`. I'd be inclined to just inline this proof into the place you need it